# Import Library and csv files

All the data is collected from google trend

In [1]:
import pandas as pd

In [2]:
fb = pd.read_csv(r'/Users/lih-anhsu/Desktop/Social Media/fb.csv')

fb_geo = pd.read_csv(r'/Users/lih-anhsu/Desktop/Social Media/fb_geo.csv')

friendster = pd.read_csv(r'/Users/lih-anhsu/Desktop/Social Media/friendster.csv')

friendster_geo = pd.read_csv(r'/Users/lih-anhsu/Desktop/Social Media/friendster_geo.csv')

google_plus = pd.read_csv(r'/Users/lih-anhsu/Desktop/Social Media/Google+.csv')

google_plus_geo = pd.read_csv(r'/Users/lih-anhsu/Desktop/Social Media/Google+_geo.csv')

hi5 = pd.read_csv(r'/Users/lih-anhsu/Desktop/Social Media/Hi5.csv')

hi5_geo = pd.read_csv(r'/Users/lih-anhsu/Desktop/Social Media/Hi5_geo.csv')

ig = pd.read_csv(r'/Users/lih-anhsu/Desktop/Social Media/ig.csv')

ig_geo = pd.read_csv(r'/Users/lih-anhsu/Desktop/Social Media/ig_geo.csv')

linkedin = pd.read_csv(r'/Users/lih-anhsu/Desktop/Social Media/Linkedin.csv')

linkedin_geo = pd.read_csv(r'/Users/lih-anhsu/Desktop/Social Media/Linkedin_geo.csv')

myspace = pd.read_csv(r'/Users/lih-anhsu/Desktop/Social Media/Myspace.csv')

myspace_geo = pd.read_csv(r'/Users/lih-anhsu/Desktop/Social Media/Myspace_geo.csv')

orkut = pd.read_csv(r'/Users/lih-anhsu/Desktop/Social Media/Orkut.csv')

orkut_geo = pd.read_csv(r'/Users/lih-anhsu/Desktop/Social Media/Orkut_geo.csv')

reddit = pd.read_csv(r'/Users/lih-anhsu/Desktop/Social Media/Reddit.csv')

reddit_geo = pd.read_csv(r'/Users/lih-anhsu/Desktop/Social Media/Reddit_geo.csv')

snapchat = pd.read_csv(r'/Users/lih-anhsu/Desktop/Social Media/snapchat.csv')

snapchat_geo = pd.read_csv(r'/Users/lih-anhsu/Desktop/Social Media/snapchat_geo.csv')

threads = pd.read_csv(r'/Users/lih-anhsu/Desktop/Social Media/Threads.csv')

threads_geo = pd.read_csv(r'/Users/lih-anhsu/Desktop/Social Media/Threads_geo.csv')

tiktok = pd.read_csv(r'/Users/lih-anhsu/Desktop/Social Media/TikTok.csv')

tiktok_geo = pd.read_csv(r'/Users/lih-anhsu/Desktop/Social Media/TikTok_geo.csv')

tumblr = pd.read_csv(r'/Users/lih-anhsu/Desktop/Social Media/Tumblr.csv')

tumblr_geo = pd.read_csv(r'/Users/lih-anhsu/Desktop/Social Media/Tumblr_geo.csv')

twitter = pd.read_csv(r'/Users/lih-anhsu/Desktop/Social Media/twitter.csv')

twitter_geo = pd.read_csv(r'/Users/lih-anhsu/Desktop/Social Media/twitter_geo.csv')

vine = pd.read_csv(r'/Users/lih-anhsu/Desktop/Social Media/Vine.csv')

vine_geo = pd.read_csv(r'/Users/lih-anhsu/Desktop/Social Media/Vine_geo.csv')

weibo = pd.read_csv(r'/Users/lih-anhsu/Desktop/Social Media/weibo.csv')

weibo_geo = pd.read_csv(r'/Users/lih-anhsu/Desktop/Social Media/weibo_geo.csv')

yt = pd.read_csv(r'/Users/lih-anhsu/Desktop/Social Media/Youtube.csv')

yt_geo = pd.read_csv(r'/Users/lih-anhsu/Desktop/Social Media/Youtube_geo.csv')

# Exploratory Data Analysis & Data Cleaning

merge above df into two df: one for geo related df, another for trend of social media df

## trend df

In [3]:
fb.head()

Category: All categories
Month      Facebook: (Worldwide)
2004-01                        0
2004-02                        0
2004-03                        0
2004-04                        0

In [4]:
friendster.head()

Category: Social Networks
Month     Friendster: (Worldwide)
2004-01                         4
2004-02                         6
2004-03                         5
2004-04                         5

Packing multiple df into a list reset index for each df

convert first row into column header

convert 'Month' from object into date type, delete': (Worldwide)'

In [5]:
all_df = [fb, friendster, google_plus, hi5, ig, linkedin, myspace, orkut, reddit, snapchat, threads, tiktok, tumblr, twitter, vine, weibo, yt]

In [6]:
for i in range(len(all_df)):
    all_df[i].reset_index(inplace=True)
    all_df[i].columns = all_df[i].iloc[0] #replace column header by first row
    all_df[i] = all_df[i].drop(all_df[i].index[0]) #drop first row
    all_df[i].rename(columns={'Month': 'Date(year_month)'}, inplace=True)
    all_df[i].columns = all_df[i].columns.str.strip().str.replace(': (Worldwide)', '', regex=False)
    print(all_df[i])


0   Date(year_month) Facebook
1            2004-01        0
2            2004-02        0
3            2004-03        0
4            2004-04        0
5            2004-05        0
..               ...      ...
231          2023-03        4
232          2023-04        4
233          2023-05        4
234          2023-06        4
235          2023-07        4

[235 rows x 2 columns]
0   Date(year_month) Friendster
1            2004-01          4
2            2004-02          6
3            2004-03          5
4            2004-04          5
5            2004-05          5
..               ...        ...
231          2023-03         <1
232          2023-04         <1
233          2023-05         <1
234          2023-06         <1
235          2023-07         <1

[235 rows x 2 columns]
0   Date(year_month) Google+
1            2009-10       0
2            2009-11       0
3            2009-12       0
4            2010-01       0
5            2010-02      <1
..               ...     ...
162  

In here, notice google+ and weibo (both appear later than others) have <235 dates

Unpack the list into individual df

In [7]:
fb, friendster, google_plus, hi5, ig, linkedin, myspace, orkut, reddit, snapchat, threads, tiktok, tumblr, twitter, vine, weibo, yt = all_df

In [8]:
#check one of the df
hi5.head()

0 Date(year_month) Hi5
1          2004-01  <1
2          2004-02  <1
3          2004-03  <1
4          2004-04  <1
5          2004-05   1

Add 'social_platform' column, rename second column as 'search trend count', union all df, replace '<1' by 0.5, convert Date from object into date dtype

In [9]:
social_platform = fb.columns[1] #get the second column header

fb.insert(2, 'Social Media Platform', social_platform) #insert the header as new column content

fb.head()

0 Date(year_month) Facebook Social Media Platform
1          2004-01        0              Facebook
2          2004-02        0              Facebook
3          2004-03        0              Facebook
4          2004-04        0              Facebook
5          2004-05        0              Facebook

In [10]:
# friendster
social_platform = friendster.columns[1] 

friendster.insert(2, 'Social Media Platform', social_platform)

# google_plus
social_platform = google_plus.columns[1] 

google_plus.insert(2, 'Social Media Platform', social_platform)

# hi5
social_platform = hi5.columns[1] 

hi5.insert(2, 'Social Media Platform', social_platform)

# ig
social_platform = ig.columns[1] 

ig.insert(2, 'Social Media Platform', social_platform)

# linkedin
social_platform = linkedin.columns[1] 

linkedin.insert(2, 'Social Media Platform', social_platform)

# myspace
social_platform = myspace.columns[1] 

myspace.insert(2, 'Social Media Platform', social_platform)

# orkut
social_platform = orkut.columns[1] 

orkut.insert(2, 'Social Media Platform', social_platform)

# reddit
social_platform = reddit.columns[1] 

reddit.insert(2, 'Social Media Platform', social_platform)

# snapchat
social_platform = snapchat.columns[1] 

snapchat.insert(2, 'Social Media Platform', social_platform)

# threads
social_platform = threads.columns[1] 

threads.insert(2, 'Social Media Platform', social_platform)

# tiktok
social_platform = tiktok.columns[1] 

tiktok.insert(2, 'Social Media Platform', social_platform)

# tumblr
social_platform = tumblr.columns[1] 

tumblr.insert(2, 'Social Media Platform', social_platform)

# twitter
social_platform = twitter.columns[1] 

twitter.insert(2, 'Social Media Platform', social_platform)

# vine
social_platform = vine.columns[1] 

vine.insert(2, 'Social Media Platform', social_platform)

# weibo
social_platform = weibo.columns[1] 

weibo.insert(2, 'Social Media Platform', social_platform)

# yt
social_platform = yt.columns[1] 

yt.insert(2, 'Social Media Platform', social_platform)

In [11]:
#pick one to check it
twitter.head(2)

0 Date(year_month) Twitter Social Media Platform
1          2004-01      <1               Twitter
2          2004-02      <1               Twitter

In [12]:
# rename columns
fb.rename(columns = {fb.columns[1]: 'Search Times Count'}, inplace = True) #using indexing to rename column

fb.head()

0 Date(year_month) Search Times Count Social Media Platform
1          2004-01                  0              Facebook
2          2004-02                  0              Facebook
3          2004-03                  0              Facebook
4          2004-04                  0              Facebook
5          2004-05                  0              Facebook

In [13]:
#friendster
friendster.rename(columns = {friendster.columns[1]: 'Search Times Count'}, inplace = True)

#google_plus
google_plus.rename(columns = {google_plus.columns[1]: 'Search Times Count'}, inplace = True)

#hi5
hi5.rename(columns = {hi5.columns[1]: 'Search Times Count'}, inplace = True)

#ig
ig.rename(columns = {ig.columns[1]: 'Search Times Count'}, inplace = True)

#linkedin
linkedin.rename(columns = {linkedin.columns[1]: 'Search Times Count'}, inplace = True)

#myspace
myspace.rename(columns = {myspace.columns[1]: 'Search Times Count'}, inplace = True)

#orkut
orkut.rename(columns = {orkut.columns[1]: 'Search Times Count'}, inplace = True)

#reddit
reddit.rename(columns = {reddit.columns[1]: 'Search Times Count'}, inplace = True)

#snapchat
snapchat.rename(columns = {snapchat.columns[1]: 'Search Times Count'}, inplace = True)

#threads
threads.rename(columns = {threads.columns[1]: 'Search Times Count'}, inplace = True)

#tiktok
tiktok.rename(columns = {tiktok.columns[1]: 'Search Times Count'}, inplace = True)

#tumblr
tumblr.rename(columns = {tumblr.columns[1]: 'Search Times Count'}, inplace = True)

#twitter
twitter.rename(columns = {twitter.columns[1]: 'Search Times Count'}, inplace = True)

#vine
vine.rename(columns = {vine.columns[1]: 'Search Times Count'}, inplace = True)

#weibo
weibo.rename(columns = {weibo.columns[1]: 'Search Times Count'}, inplace = True)

#yt
yt.rename(columns = {yt.columns[1]: 'Search Times Count'}, inplace = True)

In [14]:
tiktok.head(2)

0 Date(year_month) Search Times Count Social Media Platform
1          2004-01                  0                TikTok
2          2004-02                  0                TikTok

union all df by concat

In [15]:
social_media_trend = pd.concat([fb, friendster, google_plus,
                                hi5, ig, linkedin,
                                myspace, orkut, reddit,
                                snapchat, threads, tiktok,
                                tumblr, twitter, vine,
                                weibo, yt], 
                                ignore_index = True)

social_media_trend

0    Date(year_month) Search Times Count Social Media Platform
0             2004-01                  0              Facebook
1             2004-02                  0              Facebook
2             2004-03                  0              Facebook
3             2004-04                  0              Facebook
4             2004-05                  0              Facebook
...               ...                ...                   ...
3852          2023-03                  6               YouTube
3853          2023-04                  6               YouTube
3854          2023-05                  6               YouTube
3855          2023-06                  6               YouTube
3856          2023-07                  6               YouTube

[3857 rows x 3 columns]

replace '<1' by 0.5, convert dtype into float and date form

In [16]:
social_media_trend['Search Times Count'].replace('<1', '0.5', inplace = True)

In [17]:
# Create a list-like object containing the value '<1'
check_value = ['<1']

check_value_count = social_media_trend['Search Times Count'].isin(check_value).sum()

check_value_count

0

In [18]:
social_media_trend['Date(year_month)'] = pd.to_datetime(social_media_trend['Date(year_month)'])

social_media_trend['Date(year_month)']

0      2004-01-01
1      2004-02-01
2      2004-03-01
3      2004-04-01
4      2004-05-01
          ...    
3852   2023-03-01
3853   2023-04-01
3854   2023-05-01
3855   2023-06-01
3856   2023-07-01
Name: Date(year_month), Length: 3857, dtype: datetime64[ns]

In [19]:
social_media_trend['Search Times Count'] = social_media_trend['Search Times Count'].astype('float')

social_media_trend['Search Times Count'].dtypes

dtype('float64')

In [20]:
# save as csv file
#social_media_trend.to_csv(r'/Users/lih-anhsu/Desktop/Social Media/social_media_trend.csv')

## geo df

the normalized df calculats the percentage of each value relative to the top 1 usage country (which is considered as 100%)

In [21]:
fb_geo.head()

Category: All categories
Country              Facebook: (1/1/04 - 7/13/23)
São Tomé & Príncipe                           100
Angola                                         95
Cape Verde                                     73
Brazil                                         42

In [22]:
friendster_geo.head()

Category: Social Networks
Country                   Friendster: (1/1/04 - 7/13/23)
Philippines                                          100
Brunei                                                59
Malaysia                                              48
Northern Mariana Islands                              41

In [23]:
google_plus_geo.head()

Category: Social Networks
Country            Google+: (9/14/09 - 7/14/23)
Monaco                                      100
St. Kitts & Nevis                            63
Liechtenstein                                46
Papua New Guinea                             35

pack all the geo df into all_df_geo list

remove 'Category: Social Networks' and convert first row into column header

In [24]:
all_df_geo = [fb_geo, friendster_geo, google_plus_geo,
              hi5_geo, ig_geo, linkedin_geo,
              myspace_geo, orkut_geo, reddit_geo,
              snapchat_geo, threads_geo, tiktok_geo,
              tumblr_geo, twitter_geo, vine_geo,
              weibo_geo, yt_geo]

In [25]:
for i in range(len(all_df_geo)):
    all_df_geo[i].reset_index(inplace=True)
    all_df_geo[i].columns = all_df_geo[i].iloc[0] #replace column header by first row
    all_df_geo[i] = all_df_geo[i].drop(all_df_geo[i].index[0]) #drop first row
    
    # now it look like this: Country | Facebook: (1/1/04 - 7/13/23)
    all_df_geo[i] = all_df_geo[i].astype('str')
    
    #keeping only the part before ':'
    all_df_geo[i].columns = all_df_geo[i].columns.to_series().str.split(':', n = 1).str[0] # n=1, ensures that the split occurs only at the first occurrence of ':' in the string
    
    # fill null with 0, but in here, the nan in second column is object, fillna() can't be apply in here
    all_df_geo[i] = all_df_geo[i].replace('nan', '0')
    
    print(all_df_geo[i])

0                Country Facebook
1    São Tomé & Príncipe      100
2                 Angola       95
3             Cape Verde       73
4                 Brazil       42
5          Guinea-Bissau       40
..                   ...      ...
246      Wallis & Futuna        0
247                Samoa        0
248               Kosovo        0
249                Yemen        0
250               Zambia        0

[250 rows x 2 columns]
0                     Country Friendster
1                 Philippines        100
2                      Brunei         59
3                    Malaysia         48
4    Northern Mariana Islands         41
5                        Guam         25
..                        ...        ...
246                   Vanuatu          0
247           Wallis & Futuna          0
248                     Samoa          0
249                    Zambia          0
250                  Zimbabwe          0

[250 rows x 2 columns]
0                   Country Google+
1               

unpack all_df_geo and add a col for social media platform label; convert second col into int dtype

In [26]:
[fb_geo, friendster_geo, google_plus_geo, hi5_geo, ig_geo, linkedin_geo,
 myspace_geo, orkut_geo, reddit_geo, snapchat_geo, threads_geo,
 tiktok_geo, tumblr_geo, twitter_geo, vine_geo, weibo_geo, yt_geo] = all_df_geo

In [27]:
fb_geo

0                Country Facebook
1    São Tomé & Príncipe      100
2                 Angola       95
3             Cape Verde       73
4                 Brazil       42
5          Guinea-Bissau       40
..                   ...      ...
246      Wallis & Futuna        0
247                Samoa        0
248               Kosovo        0
249                Yemen        0
250               Zambia        0

[250 rows x 2 columns]

In [28]:
social_platform = fb_geo.columns[1] #get the second column header

fb_geo.insert(2, 'Social Media Platform', social_platform) #insert the header as new column content

fb_geo.rename(columns = {fb_geo.columns[1]: 'Percentage %'}, inplace = True) #using indexing to rename column

fb_geo.head()

0              Country Percentage % Social Media Platform
1  São Tomé & Príncipe          100              Facebook
2               Angola           95              Facebook
3           Cape Verde           73              Facebook
4               Brazil           42              Facebook
5        Guinea-Bissau           40              Facebook

In [29]:
#friendster_geo
social_platform = friendster_geo.columns[1] 

friendster_geo.insert(2, 'Social Media Platform', social_platform) 

friendster_geo.rename(columns = {friendster_geo.columns[1]: 'Percentage %'}, inplace = True) 

#google_plus_geo
social_platform = google_plus_geo.columns[1] 

google_plus_geo.insert(2, 'Social Media Platform', social_platform) 

google_plus_geo.rename(columns = {google_plus_geo.columns[1]: 'Percentage %'}, inplace = True) 

#hi5_geo
social_platform = hi5_geo.columns[1] 

hi5_geo.insert(2, 'Social Media Platform', social_platform) 

hi5_geo.rename(columns = {hi5_geo.columns[1]: 'Percentage %'}, inplace = True) 

#ig_geo
social_platform = ig_geo.columns[1] 

ig_geo.insert(2, 'Social Media Platform', social_platform) 

ig_geo.rename(columns = {ig_geo.columns[1]: 'Percentage %'}, inplace = True) 

#linkedin_geo
social_platform = linkedin_geo.columns[1] 

linkedin_geo.insert(2, 'Social Media Platform', social_platform) 

linkedin_geo.rename(columns = {linkedin_geo.columns[1]: 'Percentage %'}, inplace = True) 

#myspace_geo
social_platform = myspace_geo.columns[1] 

myspace_geo.insert(2, 'Social Media Platform', social_platform) 

myspace_geo.rename(columns = {myspace_geo.columns[1]: 'Percentage %'}, inplace = True) 

#orkut_geo
social_platform = orkut_geo.columns[1] 

orkut_geo.insert(2, 'Social Media Platform', social_platform) 

orkut_geo.rename(columns = {orkut_geo.columns[1]: 'Percentage %'}, inplace = True) 

#reddit_geo
social_platform = reddit_geo.columns[1] 

reddit_geo.insert(2, 'Social Media Platform', social_platform) 

reddit_geo.rename(columns = {reddit_geo.columns[1]: 'Percentage %'}, inplace = True) 

#snapchat_geo
social_platform = snapchat_geo.columns[1] 

snapchat_geo.insert(2, 'Social Media Platform', social_platform) 

snapchat_geo.rename(columns = {snapchat_geo.columns[1]: 'Percentage %'}, inplace = True) 

#threads_geo
social_platform = threads_geo.columns[1] 

threads_geo.insert(2, 'Social Media Platform', social_platform) 

threads_geo.rename(columns = {threads_geo.columns[1]: 'Percentage %'}, inplace = True) 

#tiktok_geo
social_platform = tiktok_geo.columns[1] 

tiktok_geo.insert(2, 'Social Media Platform', social_platform) 

tiktok_geo.rename(columns = {tiktok_geo.columns[1]: 'Percentage %'}, inplace = True) 

#tumblr_geo
social_platform = tumblr_geo.columns[1] 

tumblr_geo.insert(2, 'Social Media Platform', social_platform) 

tumblr_geo.rename(columns = {tumblr_geo.columns[1]: 'Percentage %'}, inplace = True) 

#twitter_geo
social_platform = twitter_geo.columns[1] 

twitter_geo.insert(2, 'Social Media Platform', social_platform) 

twitter_geo.rename(columns = {twitter_geo.columns[1]: 'Percentage %'}, inplace = True) 

#vine_geo
social_platform = vine_geo.columns[1] 

vine_geo.insert(2, 'Social Media Platform', social_platform) 

vine_geo.rename(columns = {vine_geo.columns[1]: 'Percentage %'}, inplace = True) 

#weibo_geo
social_platform = weibo_geo.columns[1] 

weibo_geo.insert(2, 'Social Media Platform', social_platform) 

weibo_geo.rename(columns = {weibo_geo.columns[1]: 'Percentage %'}, inplace = True) 

#yt_geo
social_platform = yt_geo.columns[1] 

yt_geo.insert(2, 'Social Media Platform', social_platform) 

yt_geo.rename(columns = {yt_geo.columns[1]: 'Percentage %'}, inplace = True) 

In [30]:
#check on of the df
tiktok_geo

0                   Country Percentage % Social Media Platform
1                     Palau          100                TikTok
2            American Samoa           82                TikTok
3                     Egypt           79                TikTok
4     St. Pierre & Miquelon           78                TikTok
5                   Algeria           75                TikTok
..                      ...          ...                   ...
246                  Tuvalu            0                TikTok
247   U.S. Outlying Islands            0                TikTok
248            Vatican City            0                TikTok
249  British Virgin Islands            0                TikTok
250         Wallis & Futuna            0                TikTok

[250 rows x 3 columns]

Union by concat of all df and convert Percentage col into int dtype

In [31]:
social_media_user_geo = pd.concat([fb_geo, friendster_geo, google_plus_geo,
                                hi5_geo, ig_geo, linkedin_geo,
                                myspace_geo, orkut_geo, reddit_geo,
                                snapchat_geo, threads_geo, tiktok_geo,
                                tumblr_geo, twitter_geo, vine_geo,
                                weibo_geo, yt_geo], 
                                ignore_index = True)

social_media_user_geo

0                                    Country Percentage %  \
0                        São Tomé & Príncipe          100   
1                                     Angola           95   
2                                 Cape Verde           73   
3                                     Brazil           42   
4                              Guinea-Bissau           40   
...                                      ...          ...   
4245  South Georgia & South Sandwich Islands            0   
4246                                 Tokelau            0   
4247                                  Tuvalu            0   
4248                   U.S. Outlying Islands            0   
4249                            Vatican City            0   

0    Social Media Platform  
0                 Facebook  
1                 Facebook  
2                 Facebook  
3                 Facebook  
4                 Facebook  
...                    ...  
4245               YouTube  
4246               YouTube  
4247               YouTube  
4248               YouTube  
4249               YouTube  

[4250 rows x 3 columns]

In [32]:
social_media_user_geo['Percentage %'].replace('<1', '0.5', inplace = True)

In [33]:
# Create a list-like object containing the value '<1'
check_value_geo = ['<1']

check_value_geo_count = social_media_user_geo['Percentage %'].isin(check_value_geo).sum()

check_value_geo_count

0

In [34]:
social_media_user_geo['Percentage %'] = social_media_user_geo['Percentage %'].astype('float')

social_media_user_geo['Percentage %'].dtype

dtype('float64')

In [35]:
social_media_user_geo.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4250 entries, 0 to 4249
Data columns (total 3 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Country                4250 non-null   object 
 1   Percentage %           4250 non-null   float64
 2   Social Media Platform  4250 non-null   object 
dtypes: float64(1), object(2)
memory usage: 99.7+ KB


In [36]:
#save as csv file
#social_media_user_geo.to_csv(r'/Users/lih-anhsu/Desktop/Social Media/social_media_user_geo.csv')